R/upload.R

```diff
@@ -0,0 +1,18 @@
+upload_input <- function(queue) {
+  function(file, filename) {
+    md5 <- toupper(digest::digest(file, algo = "md5"))
+    dest_path <- file.path(
+      queue$uploads_dir,
+      paste0(md5, ".", tools::file_ext(filename)))
+    if (!file.exists(dest_path)) {
+      t <- tempfile()
+      writeBin(file, t)
+      file.copy(t, dest_path)
```

Comment:
This will be safe with multiple hintr instances running?

Reply:
it depends what you're trying to do. If you want an atomic write, then you need to make the tempfile be on the same filesystem as the destination, then use `file.rename`. So if `dest_path` is your upload directory, we could do

```
dest_path <- file.path(
  queue$uploads_dir,
  paste0(md5, ".", tools::file_ext(filename)))
if (!file.exists(dest_path)) {
  tmpdir <- file.path(queue$uploads_dir, ".incoming"))
  dir.create(tmpdir, FALSE, TRUE
  t <- tempfile(tmpdir = tmpdir)
  writeBin(file, t)
  file.rename(t, dest_path)
}
```

as you have it, you might as well do `writeBin(file, dest_path)` as that will be at least as fast as the later `file.copy`

Of course,  if the other copies of hint/hintr are doing a plain write you have a potential race condition, but tbh it's not that likely

Reply:
Thanks, though not likely would be good to avoid it. Have added this, thanks!